lib/lz4hc.c

```diff
@@ -583,11 +582,12 @@ LZ4_FORCE_INLINE int LZ4HC_compress_hashChain (
         }
 
         /*
-        * OK, now we have 3 ascending matches; let's write at least the first one
-        * ip & ref are known; Now for ml
+        * OK, now we have 3 ascending matches;
+        * let's write the first one ML1.
+        * ip & ref are known; Now decide ml.
         */
         if (start2 < ip+ml) {
-            if ((start2 - ip) < (int)ML_MASK) {
+            if ((start2 - ip) < OPTIMAL_ML) {
```

Comment:
Why change this?

Reply:
I'm pretty sure that's the correct intention : 
the algorithm wants to shorten matches > `OPTIMAL_ML`, in order to save an additional length byte. When length `< OPTIMAL_ML`, it's not necessary to trigger the shortener.

Note that, strangely enough, this change doesn't impact compression result, 
presumably because this check is actually redundant with some previous check, completed multiple lines above.
Which also leads me to believe this code could be better optimized.
But that's a much more complex task, and I did not want to make this patch any more complex.